ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -0,0 +1,154 @@
+            try (InputStream stream = Files.newInputStream(path)) {
+                char[] passwordChars = password.isPresent() ? password.get().toCharArray() : null;
+                keyStore.load(stream, passwordChars);
+            }
+            return keyStore;
+        } catch (GeneralSecurityException | IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    /**
+     * Return a new {@link KeyStore} that contains the key and certificate chain with the provided alias in the given
+     * key store.
+     *
+     * @param original
+     *        key store that contains the key and certificate chain
+     * @param password
+     *        the password for the provided key store. Will also be used as the password of the returned key store.
+     * @param alias
+     *        the alias of the key and certificate chain in the provided key store
+     * @return a newly constructed key store that contains a single entry that consists of the key and certificate chain
+     *         with the provided alias in the given key store. The trust store will be of the same type as the provided
+     *         one, will use the same password, and will store the key and certificate chain using the same alias.
+     * @throws IllegalStateException
+     *         if the provided key store does not contain a key and certificate chain with the given alias
+     */
+    static KeyStore newKeyStoreWithEntry(KeyStore original, String password, String alias) {
+        try {
+            KeyStore newKeyStore = KeyStore.getInstance(original.getType());
+            newKeyStore.load(null, password.toCharArray());
+
+            Key aliasKey = original.getKey(alias, password.toCharArray());
+            if (aliasKey == null) {
+                throw new IllegalStateException(
+                        String.format("Could not find key with alias \"%s\" in key store", alias));
+            }
+            Certificate[] certificateChain = original.getCertificateChain(alias);
+            if (certificateChain == null) {
+                throw new IllegalStateException(
+                        String.format("Could not find certificate chain with alias \"%s\" in key store", alias));
+            }
+
+            newKeyStore.setKeyEntry(alias, aliasKey, password.toCharArray(), certificateChain);
+            return newKeyStore;
+        } catch (GeneralSecurityException | IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private static Certificate readX509Certificate(Path certificateFilePath) {
+        try {
+            CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+            Certificate certificate;
+            try (InputStream stream = Files.newInputStream(certificateFilePath)) {
+                certificate = certFactory.generateCertificate(stream);
+            }
+            return certificate;
+        } catch (GeneralSecurityException | IOException e) {
```

Comment:
Keeping in this case to address Mark's suggestion -- allows throwing a more specific exception for error reading certificate specified by the path